### PR TITLE
#729 v3 authorization fail

### PR DESCRIPTION
**Pull request by Erwane**
Skip cakephp/authorization plugin for DebugKit requests only

### CODE REVIEW

File: tests/TestCase/Controller/DebugKitControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        Configure::write('debug', false);
+
+        $request = new ServerRequest(['url' => '/debug-kit/']);
+        $controller = new DebugKitController($request, new Response());
+        $event = new Event('testing');
+
+        // try/catch/finally instead of expectExcetion
+        // to restore `debug` configuration
+        try {
+            $controller->beforeFilter($event);
```
Comment (markstory) [inline tests/TestCase/Controller/DebugKitControllerTest.php]: Integration tests generally make requests to the application and check the response contents.

File: tests/TestCase/Controller/DebugKitControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        } catch (Exception $e) {
+            $this->assertInstanceOf('Cake\Http\Exception\NotFoundException', $e);
+            $this->assertSame('Not available without debug mode on.', $e->getMessage());
+        } finally {
+            Configure::write('debug', $oldStatus);
```
Comment (markstory) [inline tests/TestCase/Controller/DebugKitControllerTest.php]: You don't need to do this. The test harness will reset `Configure` state during its teardown method.

File: src/Controller/DebugKitController.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public function beforeFilter(Event $event)
+    {
+        // TODO add config override.
+        if (!Configure::read('debug')) {
+            throw new NotFoundException('Not available without debug mode on.');
+        }
+
+        // Skip authorization for DebuKit requests
+        $authorizationService = $this->getRequest()->getAttribute('authorization');
```
Comment (markstory) [inline src/Controller/DebugKitController.php]: I thought this was going to be a behavior that users could turn on. Something like `DebugKit.safeTLD` works.

**Comment by Erwane** [comment]
Thanks. I thought my code was clean ;)
I will do better.

File: tests/TestCase/Controller/DebugKitControllerTest.php (hunk 1 above)
Comment (Erwane) [reply tests/TestCase/Controller/DebugKitControllerTest.php]: I'm ok with you about `testDebugDisabled` but for the `testSkipAuthorization` i should test the AuthorizationService::authorizationChecked() status, and i only can do that thrue controller request.
Any ideas ?

File: src/Controller/DebugKitController.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        // If CakePHP Authorization\Authorization plugin is enabled,
+        // ignore it, only if `DebugKit.ignoreAuthorization` is set to true
+        $authorizationService = $this->getRequest()->getAttribute('authorization');
+        if ($authorizationService instanceof \Authorization\AuthorizationService) {
+            if (Configure::read('DebugKit.ignoreAuthorization')) {
+                $authorizationService->skipAuthorization();
+            } else {
+                Log::info(
```
Comment (markstory) [inline src/Controller/DebugKitController.php]: :+1: This is a nice inclusion.

**Pull request merged**